.github/workflows/clean_dockerhub_images.yml

```diff
@@ -0,0 +1,62 @@
+name: Docker Cleanup
+
+on:
+  schedule:
+    - cron: '0 12 * * 1'  # Runs every Monday at noon (UTC)
+  workflow_dispatch:
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'  # Specify the version of Python
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install requests
+
+    - name: Clean up old Docker images
+      env:
+        DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+        DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+        REPO_NAME: "scylladb/gocql-extended-ci"
+        DELETE_AFTER_DAYS: 4  # Number of days after which to delete images
+      run: |
+        python <<EOF
+        import requests
+        from datetime import datetime, timedelta
```

Comment:
Let's move all this code into a file in `scripts` or `ci` directory.
Create and `Makefile` to call it.